napi_js_props.md

```diff
@@ -4,16 +4,65 @@ N-API exposes a set of APIs to interrogate and set properties on JavaScript obje
 Some of these types are documented under [Section 7](https://tc39.github.io/ecma262/#sec-operations-on-objects) 
 of the [ECMAScript Language Specification](https://tc39.github.io/ecma262/).
 
-Fundamentally, all properties in N-API can be represented in one of the following forms:
+Fundamentally, all property keys in N-API can be represented in one of the following forms:
 - Named: a simple utf8 encoded string
 - Integer-Indexed: an index value represented by `uint32_t`
-- JavaScript Object: a complex JavaScript Object represented by `napi_value`. This is the case if the key for a particular property is not a String or Number.
+- JavaScript Object: a JavaScript Object represented by `napi_value`. This can be a `napi_value` representing a String, Number or Symbol.
 
 N-API values are represented by the type `napi_value`.
 Any N-API call that requires a JavaScript object takes in a `napi_value`.
 However, it's the caller's responsibility to make sure that the 
 `napi_value` in question is of the JavaScript type expected by the API.
 
+## Structures
+
+### *napi_property_attributes*
+
+#### Definition
+```
+typedef enum {
+  napi_default = 0,
+  napi_read_only = 1 << 0,
+  napi_dont_enum = 1 << 1,
+  napi_dont_delete = 1 << 2,
+  napi_static_property = 1 << 10,
+} napi_property_attributes;
+```
+
+### Description
+`napi_property_attributes` are flags used to control the behavior of properties set on a JavaScript object. They roughly correspond to the attributes listed in [Section 6.1.7.1](https://tc39.github.io/ecma262/#table-2) of the [ECMAScript Language Specification](https://tc39.github.io/ecma262/).
+
+`napi_default` - Used to indicate that no explicit attributes are set on the given property
```

Comment:
Maybe mention that a property is writable, enumerable, and configurable by default.